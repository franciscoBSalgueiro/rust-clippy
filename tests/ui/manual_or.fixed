#![warn(clippy::manual_or)]
#[allow(irrefutable_let_patterns)]

fn main() {
    let a = true;
    let b = false;
    let c = true;

    let _ = a || b; //~ ERROR: this `if`-then-`else` expression can be simplified with `||`

    let _ = if a {
        true
    } else if b {
        false
    } else {
        c
    };

    let _ = a || !b;
    //~^ ERROR: this `if`-then-`else` expression can be simplified with `||`

    let _ = !a || b;
    //~^ ERROR: this `if`-then-`else` expression can be simplified with `||`

    let _ = if let x = a { true } else { b };
}
