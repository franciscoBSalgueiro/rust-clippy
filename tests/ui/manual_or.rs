#![warn(clippy::manual_or)]
#[allow(irrefutable_let_patterns)]

fn main() {
    let a = true;
    let b = false;
    let c = true;

    let _ = if a { true } else { b }; //~ ERROR: this `if`-then-`else` expression can be simplified with `||`

    let _ = if a {
        true
    } else if b {
        false
    } else {
        c
    };

    let _ = if a { true } else { !b };
    //~^ ERROR: this `if`-then-`else` expression can be simplified with `||`

    let _ = if !a { true } else { b };
    //~^ ERROR: this `if`-then-`else` expression can be simplified with `||`

    let _ = if let x = a { true } else { b };
}
