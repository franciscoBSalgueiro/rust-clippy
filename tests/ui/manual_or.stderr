error: this `if`-then-`else` expression can be simplified with `||`
  --> tests/ui/manual_or.rs:9:13
   |
LL |     let _ = if a { true } else { b };
   |             ^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `a || b`
   |
   = note: `-D clippy::manual-or` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::manual_or)]`

error: this `if`-then-`else` expression can be simplified with `||`
  --> tests/ui/manual_or.rs:19:13
   |
LL |     let _ = if a { true } else { !b };
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `a || !b`

error: this `if`-then-`else` expression can be simplified with `||`
  --> tests/ui/manual_or.rs:22:13
   |
LL |     let _ = if !a { true } else { b };
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `!a || b`

error: aborting due to 3 previous errors

